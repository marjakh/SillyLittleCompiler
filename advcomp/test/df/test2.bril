@main() {
.b1:
  a: int = const 47;
  b: int = const 42;
  jmp .b2;
.b2:
  a: int = const 100;
  c: int = const 5;
  jmp .b3;
.b3:
  print a;
  print b;
  print c;
}
