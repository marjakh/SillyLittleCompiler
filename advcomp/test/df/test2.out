{dest: a, op: const, type: int, value: 47}
{dest: b, op: const, type: int, value: 42}
{op: jmp, labels: ['b2']}
{dest: a, op: const, type: int, value: 100}
{dest: c, op: const, type: int, value: 5}
{op: jmp, labels: ['b3']}
{args: ['a'], op: print}
Reachable definitions for a: {{dest: a, op: const, type: int, value: 100}}
{args: ['b'], op: print}
Reachable definitions for b: {{dest: b, op: const, type: int, value: 42}}
{args: ['c'], op: print}
Reachable definitions for c: {{dest: c, op: const, type: int, value: 5}}
